clang/lib/CIR/CodeGen/CIRGenStmt.cpp

```diff
@@ -479,6 +467,11 @@ mlir::LogicalResult CIRGenFunction::buildReturnStmt(const ReturnStmt &S) {
   assert(!UnimplementedFeature::requiresReturnValueCheck());
   auto loc = getLoc(S.getSourceRange());
 
+  // TODO: Rewrite the logic to handle ReturnStmt inside SwitchStmt, then
+  //       clean up the code below.
+  if (currLexScope->IsInsideCaseNoneStmt)
+    return mlir::success();
```

Comment:
I found many sample code that failed due to incorrect terminator in block, e.g.
```
  switch(a) {
  case 0:
    break; 
    int x = 1;
  }
```
```
  switch(a) {
  case 0:
    return 0;
    return 1;
    int x = 1;
  }
```
```
for (;;) {
  break;
  int x = 1;
}
```
Looks like it's another large work, so I just skip ReturnStmt here.

Reply:
Right, can you file a new issue and list these?

Reply:
I'm opposed to `return mlir::success();` because it will just silently skips something we don't know how to handle, I rather these things fails or crash, so that it's clear that they aren't implemented? What happens when you remove this return?